tools/c7n_oci/c7n_oci/provider.py

```diff
@@ -21,13 +47,68 @@ class OCI(Provider):
     resource_map = ResourceMap
 
     def initialize(self, options):
+        if options["account_id"] is None:
+            options["oci_config"] = get_oci_profile_session(options).get_config()
         return options
 
+    def _get_subscribed_region(self, options):
+        client = get_oci_profile_session(options).client("oci.identity.IdentityClient")
+        return client.list_region_subscriptions(tenancy_id=options["oci_config"]["tenancy"]).data
```

Comment:
move the ready check here so it just returns valid regions, so further use doesn't have to check values returned.

Reply:
Done